ksql-execution/src/main/java/io/confluent/ksql/execution/plan/StreamSelectKey.java

```diff
@@ -15,21 +15,24 @@
 package io.confluent.ksql.execution.plan;
 
 import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.GenericRow;
 import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.streams.kstream.KStream;
 
 @Immutable
-public class StreamSelectKey<S> implements ExecutionStep<S> {
+public class StreamSelectKey<K> implements ExecutionStep<KStream<Struct, GenericRow>> {
```

Comment:
Do we need `K` here?  Can we not just use `ExecutionStep<KStream<?, GenericRow>>` for the `source ` param / field

Reply:
You can't pass an `ExecutionStep<KStream<K, GenericRow>>`, which is what `SchemaKStream` has, to `ExecutionStep<KStream<?, GenericRow>>`. We could make the type `ExecutionStep<? extends KStream<?, GenericRow>>`, but this felt easier to read.